driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -664,16 +713,7 @@ public void onException(Connection connection, Exception exception, long latency
 
                 @Override
                 public boolean onTimeout(Connection connection, long latency, int retryCount) {
-                    QueryState queryState = queryStateRef.get();
-                    if (!queryState.isInProgressAt(retryCount) ||
-                        !queryStateRef.compareAndSet(queryState, queryState.complete())) {
-                        logger.debug("onTimeout triggered but the response was completed by another thread, cancelling (retryCount = {}, queryState = {}, queryStateRef = {})",
-                            retryCount, queryState, queryStateRef.get());
-                        return false;
-                    }
-                    connection.release();
-                    logError(connection.address, new DriverException("Timeout waiting for response to prepare message"));
-                    retry(false, null);
+                    SpeculativeExecution.this.onTimeout(connection, latency, retryCount);
                     return true;
```

Comment:
My comment about PREPARE being idempotent also applies here, so we might want to keep a separate implementation that does not use the retry policy.

If on the contrary we keep calling the method of the enclosing class, we want to return its result instead of always returning `true`.